### PR TITLE
[INFRA-102] Adding archives.jenkins-ci.org

### PR DESCRIPTION
**Pull request by kohsuke**
... to serve historical archived releases. See [INFRA-102](https://issues.jenkins-ci.org/browse/INFRA-102) for the context.

### CODE REVIEW

**Comment by rtyler** [comment]
I'll take a look at this tomorrow, got caught up with work :(

**Comment by kohsuke** [comment]
Retriggering the pull request build.

**Comment by rtyler** [comment]
```
Failures:

  1) profile::archives should contain Package[apache2]
     Failure/Error: should contain_package('apache2')
       expected that the catalogue would contain Package[apache2]
     # ./spec/classes/profile/archives_spec.rb:6:in `block (2 levels) in <top (required)>
```

This looks like a legit test failure

**Comment by kohsuke** [comment]
My bad, that was a test problem.

**Comment by kohsuke** [comment]
@rtyler , while you review the change, please check the sanity of the expression `str2bool("${vagrant}")`

This is copied from a Google search result, and I don't understand why one just wouldn't simply do `str2bool($vagrant)`

**Comment by kohsuke** [comment]
Not sure if failure in build #174 is transient. Trying it again.

File: dist/profile/manifests/archives.pp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    exec { 'create /tmp/xvdb':
+      command => 'dd if=/dev/zero of=/tmp/xvdb bs=1M count=16; losetup /dev/loop0; losetup /dev/loop0 /tmp/xvdb',
+      unless  => 'test -f /tmp/xvdb',
+      path    => '/usr/bin:/usr/sbin:/bin:/sbin',
+      before  => Physical_volume['/dev/loop0'],
+    }
+  }
+
+  # volume configuration is in hiera
+  include ::lvm
```
Comment (rtyler) [inline dist/profile/manifests/archives.pp:21]: For consistency and style, includes should be at the top of the class. I'll tidy this up later

File: dist/profile/manifests/archives.pp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+
+
+
+  file { '/var/log/apache2/archives.jenkins-ci.org':
+    ensure => directory,
+  }
+
+  include apache
```
Comment (rtyler) [inline dist/profile/manifests/archives.pp:36]: Ditto here

**Comment by rtyler** [comment]
Mostly looks good to me, I'm going to merge this to `staging` and then do some minor non-functional cleanup before pushing to production later this afternoon.

I'm really impressed with this though!

**Pull request merged**